### ISSUE TITLE
RSA.generate() does not generate random key pair

### ISSUE THREAD

**Issue by jeroenlammersma** [Legrandin/pycryptodome#583]
When I call the generate function on RSA, I get exactly the same key pair when It's run. It supposed to generate a random pair, but somehow it is deterministic.

When I call the following:
```RSA.generate(2048).publickey().exportKey('PEM').decode('utf-8')```

I get this public key every time:
```
-----BEGIN PUBLIC KEY-----
MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAuF7TJa1y/dkakRJoYm9g
Pavb3nNyhrqM4FWSh59wUzKTZ0zhjQR/sDsgF3iuKs1XWZ6w+OQMvCRpKXpXDUAk
c8AW8wwNvMhGPWPyA2o2VdPbrTdFT2oq10IAt5nu17NVj+RAYPoOUeacRlLPrkF8
TCOw+fyfWk/s8764Q6ozBez8ejtIBXqiyE+RD9A5GKZ53357CnQjoveYIvvI0fL9
sbUsyAVl4H/k8IVriWYOdRTTpnpkWs6G8SmbkiMrv2m9ZQ69BkEaJmsX1ZaDR+nJ
YRwrQMZydVZiOrTFyzsxQhFO+4cU6UlIgDuks7D1bDzi3WBt0mRRsggI1bsGpCCS
0wIDAQAB
-----END PUBLIC KEY-----
```

I ran it on version 3.10.1 on Windows and 3.9.9 on Ubuntu, but the results stay the same.

**Comment by Legrandin** [comment]
Can you please provide more information about any of the platforms?
I definitely get different (random) RSA keys in both cases.

**Comment by jeroenlammersma** [comment]
I found the cause of the issue! It's not something related pycryptodome (fortunately). I wrote the the following code:

```
def __init__(self, key_pair = RSA.generate(2048)):
    self.key_pair = key_pair
```

However, I did not know Python evaluated the default argument only once, therefore generating the exact same key pair everytime.

_"Python’s default arguments are evaluated once when the function is defined, not each time the function is called (like it is in say, Ruby). This means that if you use a mutable default argument and mutate it, you will and have mutated that object for all future calls to the function as well."_

source: https://docs.python-guide.org/writing/gotchas/
